app/vue/src/client/preview/render.ts

```diff
@@ -48,6 +49,7 @@ export default function render({
     root[COMPONENT] = element;
   }
 
+  // @ts-ignore I'm not sure why we need this, but types are incompatible
   root[VALUES] = element.options[VALUES];
```

Comment:
the whole `VALUES` stuff is a workaround to make knobs work. each time a knob or story changes, this render function is called and also an entirely new component is created, when `const element = storyFn();` is called above. the thing is, that we dont want to replace the component, when a knob value changes, we just want to change some props based on the new knob values. only when the story changes we want to replace the component (see above `root[COMPONENT] = element;`). as it happens to be, `forceRender` is `true`, if just the knob values change (which is quite counterintuitive).

the problem is that we have to do `const element = storyFn();` in order to update knobs and get their values. thats where the `prepare` function of the story decoration kicks in, which is executed on all components returned from the story or story decorator functions and collects the default values of their props (which contain the new knob value, when we declare our knob like `{ props: { myProp: { default: TEXT_KNOB(...) } } }`). this way we can access all of these values in the component returned from the render story function: `storyFn().options[VALUES]`. in addition to that the `prepare` function also decorates all components returned from the story or story decorator functions with a functional component that passes (and therefore binds) the data from the root component's `[VALUES]` data (which is reactive) to the decorated component via props (https://github.com/storybookjs/storybook/pull/7578/files#diff-cf1aa09935b730d1855cb4c973861de0R46). so each time we change `root[VALUES]`, vue's reactivity kicks in and rerenders all components that use this data, which is basically all components in our case, because the functional component decorators of the `prepare` function access the whole data object by doing `...parent.$root[VALUES]`, which is still cheap in vue, because we dont create new component instances (that arent necessary), like when we would do `root[COMPONENT] = storyFn()` (which would also work).